CONTRIBUTING.md

```diff
@@ -89,6 +89,33 @@ TURBO_BINARY_PATH=~/repos/vercel/turbo/cli/turbo.exe npm link turbo
 
 If you're using a different package manager replace npm accordingly.
 
+## Manually testing `turbo`
+
+Before releasing, it's recommended to test the `turbo` binary manually.
+Here's a checklist of testing strategies to cover:
+
+- Test `login`, `logout`, `login --sso-team`, `link`, `unlink`
+- Test every command with `--skip-infer`.
+- Test `--dry-run` and `--graph`.
+- Test with and without daemon.
+
+There are also multiple installation scenarios worth testing:
+
+- Global-only. `turbo` is installed as global binary, no local `turbo` in repository.
+- Local-only. `turbo` is installed as local binary, no global `turbo` in PATH. turbo` is invoked via a root package script.
+- Global + local. `turbo` is installed as global binary, and local `turbo` in repository. Global `turbo` delegates to local `turbo`
+- npx. `turbo` is invoked via `npx turbo`.
+- npx + local. `turbo` is invoked via `npx turbo` in a repository with a local `turbo` binary.
```

Comment:
Do these scenarios need to be tested separately?
- If there isn't a local install, then `npx turbo` is the same as `turbo` (assuming `PATH` is correctly setup) which should be the same codepath as global only turbo
- If there's a local install then `npx turbo` goes down the same code path as local only since `npx turbo` invokes the local binary directly even if there's a global turbo present

Reply:
Hmm I guess the only difference with `npx turbo` on a local install is that the local turbo is also doing the inference and calling itself.

Reply:
That does seem worth testing actually. The first situation might not be worth testing though

Reply:
I'm pretty sure that's the same codepath as local only where turbo is invoked via a package script.